### PR TITLE
Add IFTTT Maker Channel action

### PR DESCRIPTION
**Pull request by vpolouchkine**

### CODE REVIEW

File: fastlane/lib/fastlane/actions/ifttt.rb [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+      #####################################################
+      # @!group Documentation
+      #####################################################
+
+      def self.description
+        "Connect to the IFTTT Maker Channel. https://ifttt.com/maker"
+      end
+
+      def self.available_options
+        [FastlaneCore::ConfigItem.new(key: :api_key,
```
Comment (KrauseFx) [inline fastlane/lib/fastlane/actions/ifttt.rb]: Please add a newline here

File: fastlane/lib/fastlane/actions/ifttt.rb [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+                                      is_string: true),
+          FastlaneCore::ConfigItem.new(key: :value3,
+                                       env_name: "IFTTT_VALUE3",
+                                       description: "Extra data sent with the event",
+                                       optional:true,
+                                       is_string: true)]
+      end
+
+      def self.is_supported?(platform)
+        [:ios, :mac, :android].include?(platform)
```
Comment (KrauseFx) [inline fastlane/lib/fastlane/actions/ifttt.rb]: This should be just `true`

File: fastlane/lib/fastlane/actions/ifttt.rb [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+      end
+
+      def self.authors
+        "Val Polouchkine"
```
Comment (KrauseFx) [inline fastlane/lib/fastlane/actions/ifttt.rb]: This should be `["username"]` instead

File: fastlane/lib/fastlane/actions/ifttt.rb [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+
+        req.set_form_data({
+          "value1" => options[:value1],
+          "value2" => options[:value2],
+          "value3" => options[:value3],
+        })
+
+        response = https.request(req)
+
+        raise UI.error "Failed to make a request to IFTTT. #{response.message}." unless response.code == "200"
```
Comment (KrauseFx) [inline fastlane/lib/fastlane/actions/ifttt.rb]: Remove the `raise` here

Comment (KrauseFx) [inline fastlane/lib/fastlane/actions/ifttt.rb]: It should be `UI.user_error!("lkjsdflkjfd")`

File: fastlane/lib/fastlane/actions/ifttt.rb [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+        uri = URI.parse("https://maker.ifttt.com/trigger/#{options[:event_name]}/with/key/#{options[:api_key]}")
+        https = Net::HTTP.new(uri.host, uri.port)
+        https.use_ssl = true
+
+        req = Net::HTTP::Post.new(uri.request_uri)
+
+        req.set_form_data({
+          "value1" => options[:value1],
+          "value2" => options[:value2],
+          "value3" => options[:value3],
```
Comment (KrauseFx) [inline fastlane/lib/fastlane/actions/ifttt.rb]: Is that a hard limit from IFTTTT? The number 3

Comment (vpolouchkine) [inline fastlane/lib/fastlane/actions/ifttt.rb]: Yes

**Comment by KrauseFx** [comment]
The tests seem to fail https://circleci.com/gh/fastlane/fastlane/97

**Comment by vpolouchkine** [comment]
:eyes:

**Comment by KrauseFx** [comment]
Looks good to me :+1:

**Pull request merged**